MessageBuilder.js

```diff
@@ -0,0 +1,98 @@
+class MessageBuilder {
+  setBold (val) { this.bold = val; return this }
+  setItalic (val) { this.italic = val; return this }
+  setUnderlined (val) { this.underlined = val; return this }
+  setStrikethrough (val) { this.strikethrough = val; return this }
+  setObfuscated (val) { this.obfuscated = val; return this }
+  setColor (val) { this.color = val; return this }
+  setInsertion (val) { this.insertion = val; return this }
+  setText (val) { this.text = val; return this }
+  setClickEvent (action, value) { this.clickEvent = { action, value }; return this }
+  setHoverEvent (action, type, id, name) {
+    this.hoverEvent = { action, contents: { type, id, name: { text: name } } }
+    return this
+  }
+
+  addTranslateInsertion (val) {
+    if (!this.with) this.with = []
+    this.with.push(val)
+    return this
+  }
+
+  addComponent (val) {
+    if (!this.extra) this.extra = []
+    this.extra.push(val)
+    return this
+  }
+
+  setTranslate (val) { this.translate = val; return this }
+}
+
+const supportedColors = {
+  0: 'black',
+  1: 'dark_blue',
+  2: 'dark_green',
+  3: 'dark_aqua',
+  4: 'dark_red',
+  5: 'dark_purple',
+  6: 'gold',
+  7: 'gray',
+  8: 'dark_gray',
+  9: 'blue',
+  a: 'green',
+  b: 'aqua',
+  c: 'red',
+  d: 'light_purple',
+  e: 'yellow',
+  f: 'white',
+  k: 'obfuscated',
+  l: 'bold',
+  m: 'strikethrough',
+  n: 'underlined',
+  o: 'italic',
+  r: 'reset'
+}
+
+// convert classic minecraft color strings to the message class
+function convertColorCodes (str) {
+  let lastObj = null
+  let currString = ''
+  for (let i = str.length - 1; i > -1; i--) {
+    if (str[i] !== '&') currString += str[i]
+    else if (str[i] === '&') {
+      const text = currString.split('').reverse()// reverse
+      const color = supportedColors[text.shift()]
+      const newObj = new MessageBuilder()
+      if (color === 'reset') {
+        newObj.setObfuscated(false)
```

Comment:
It's a new object, is all this necessary? Or does undefined have a different meaning? If so, you can add a reset() method

Reply:
undefined means something different since we are just trying to break inheritance of any of these properties. The code isn't used again so I don't know if its worth it to make it a function.